crates/ibc/src/applications/transfer/acknowledgement.rs

```diff
@@ -56,17 +56,21 @@ impl Display for TokenTransferAcknowledgement {
 
 impl From<TokenTransferAcknowledgement> for Vec<u8> {
     fn from(ack: TokenTransferAcknowledgement) -> Self {
+        // WARNING: Make sure all branches always return a non-empty vector.
+        // Otherwise, the conversion to `Acknowledgement` will panic.
```

Comment:
I'm also not a fan of needing that comment. Note also that I added a test to check that this conversion doesn't return a non-empty vector.

The fix would look like:
```rust
impl From<TokenTransferAcknowledgement> for NonEmptyVec<u8> {
    // ...
}

impl From<TokenTransferAcknowledgement> for Acknowledgement {
    fn from(ack: TokenTransferAcknowledgement) -> Self {
        let v: NonEmptyVec<u8> = ack.into();

        // this can be an `into()` because the only possible error case
        // (empty vector) is not possible with `NonEmptyVec<u8>`
        v.into()
    }
}
```

Reply:
Created #389